lib/robots/dor_repo/gis_assembly/package_data.rb

```diff
@@ -33,45 +35,45 @@ def generate_data_zip(rootdir)
           Dir.chdir(tmpdir)
           File.umask(002)
 
+          filename = find_metadata_file
+          raise "package-data: #{bare_druid} cannot locate metadata in temp" if filename.nil?
+
+          filenames = build_file_list(filename)
```

Comment:
This is another relatively straightforward refactor of the big nested if originally lines 37~60ish that would have required another level of testing. Abstracting this into `build_file_list` and `find_metadata_file` (line 38) made a lot of sense to me.